### ISSUE TITLE
Lines in API reference XML files are too long

### ISSUE THREAD

**Issue by svick** [dotnet/docs#1953]
The lines in XML files in API reference are way too long. This causes several fairly minor, but noticeable annoyances to contributors like me:

* When viewing the source on GitHub, I have to keep scrolling horizontally to read it.
* When viewing PRs or diffs, I have to switch to Split mode to see the differences without horizontal scrolling.
* I *think* that long lines mean that GitHub does not highlight which words changed (see e.g. the diff for https://github.com/dotnet/docs/pull/1930 and try to figure out what changed in remarks; compare it with https://github.com/svick/core-docs/commit/24815ef579a171f8a9913625148ee7ef0743afbb).

My questions (similar to those in https://github.com/dotnet/docs/issues/1951):

* Is this something that's worth fixing (considering it would change many files)?
* Would a PR by me be welcome, or do you have a better way to deal with this?
* Should it be fixed all at the same time, or would some kind of gradual process make more sense?
* Would it help to fix this in the same PR as https://github.com/dotnet/docs/issues/1951?

I have a work-in-progress code that would fix this [here](https://github.com/svick/ApiReferenceCleanup/blob/master/Program.cs). In case this is deemed worth doing, I would finish it and submit a PR.

**Comment by mairaw** [comment]
That's an interesting issue. I think before you invest any more of your time, let me discuss this with the rest of the team. I did have to manually scan the text for PR #1930 to spot the differences.

/cc @BillWagner @rpetrusha @wh-alice @dend @yishengjin1413

**Comment by GuardRex** [comment]
Just plz keep in mind a few things:
1. When you set line lengths, you're setting them permanently everywhere, for everyone, and forever. If you work in an IDE, like I do, I usually need to scroll vertically twice as much to compensate for all of these returns all over the place. I prefer to set my window (or IDE pane) to the width of text that I want to see and have the text flow into that space. My preferred width is at least twice what I commonly see in docs where returns have been introduced. When I need to access text on a mobile device that can't handle the pre-determined line length, then that introduces half-lines all over the place.
2. If GH is having a prob with unified mode and highlighting diffs, then that's an issue with GH ... not with docs. If the primary prob is GH, then I suggest a shout out to GH MCP with a feature request to fix GH before taking any drastic actions. A simple soft wrap feature would be a great benefit the entire GH universe.

**Comment by svick** [comment]
@GuardRex Yeah, if there's a good chance GitHub's UI will be improved in a reasonable timeframe, that is the best solution.

Otherwise, when it comes to line length, I was thinking of setting it to the highest value, which works on GitHub without scrolling, which seems to be about 120.

**Comment by GuardRex** [comment]
Display of text in the code files relies on HTML's `<pre>`-style formatting. For example, they use ...
```css
.blob-code-inner { white-space: pre }
```
... which results in lines not breaking. One can hack it, such as by using something like [Stylebot](https://chrome.google.com/webstore/detail/stylebot/oiaejidbmkiecgbjeifoejpgmdaleoha?hl=en) to override the GH style and set ...
```css
.blob-code-inner { white-space: pre-wrap; word-wrap: break-word; }
```
That one simple hack wraps many lines. In XML, the hack isn't complete effective: It still won't work inside of `value` attribute values. Fortunately, the hack **does** seem to work with diffs in unified view. I think with some additional CSS Ninja Tricks:tm: one should be able to have a 100% effective CSS-based workaround. Even if it works tho, I still wish GH would do more for everyone.

RE: Those lines not highlighting changes in diffs. Yeah, that one's on them. Seems like their algorithm that applies the styles, such as `.blob-code-addition`, just isn't smart enough to catch them all ... or perhaps they do shut it off on very long lines. I think that one should be a bug report to GH.
